src/extension/op_def.rs

```diff
@@ -413,4 +415,113 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn binary_polyfunc() -> Result<(), Box<dyn std::error::Error>> {
+        struct SigFun();
+        impl CustomSignatureFunc for SigFun {
+            fn compute_signature(
+                &self,
+                _name: &SmolStr,
+                arg_values: &[TypeArg],
+                _misc: &std::collections::HashMap<String, serde_yaml::Value>,
+                _exts: &ExtensionRegistry,
+            ) -> Result<PolyFuncType, SignatureError> {
+                const TP: TypeParam = TypeParam::Type(TypeBound::Any);
+                let [TypeArg::BoundedNat { n }] = arg_values else {
+                    return Err(SignatureError::InvalidTypeArgs);
+                };
+                let n = *n as usize;
+                let tvs: Vec<Type> = (0..n)
+                    .map(|_| Type::new_var_use(0, TypeBound::Any))
+                    .collect();
+                Ok(PolyFuncType::new(
+                    vec![TP],
+                    FunctionType::new(tvs.clone(), vec![Type::new_tuple(tvs)]),
+                ))
+            }
+        }
+        let mut e = Extension::new(EXT_ID);
+        let def = e.add_op_custom_sig_simple(
+            "MyOp".into(),
+            "".to_string(),
+            vec![TypeParam::max_nat()],
+            SigFun(),
+        )?;
+
+        // Base case, no type variables:
+        let args = [TypeArg::BoundedNat { n: 3 }, USIZE_T.into()];
+        assert_eq!(
+            def.compute_signature(&args, &PRELUDE_REGISTRY),
+            Ok(FunctionType::new(
+                vec![USIZE_T; 3],
+                vec![Type::new_tuple(vec![USIZE_T; 3])]
+            ))
+        );
+        assert_eq!(def.validate_args(&args, &PRELUDE_REGISTRY, &[]), Ok(()));
+
+        // Second arg may be a variable (substitutable)
+        let tyvar = Type::new_var_use(0, TypeBound::Eq);
+        let tyvars: Vec<Type> = (0..3).map(|_| tyvar.clone()).collect();
```

Comment:
fyi
```suggestion
        let tyvars = vec![tyvar.clone(); 3];
```

Reply:
Ah, thank you! Thought I'd tried that several times and not got it to work, but obviously I did something not quite right. Done :)